test/extensions/transport_sockets/tls/cert_validator/timed_cert_validator.cc

```diff
@@ -14,14 +14,17 @@ namespace Tls {
 ValidationResults TimedCertValidator::doVerifyCertChain(
     STACK_OF(X509)& cert_chain, Ssl::ValidateResultCallbackPtr callback,
     const Network::TransportSocketOptionsConstSharedPtr& transport_socket_options, SSL_CTX& ssl_ctx,
-    const CertValidator::ExtraValidationContext& /*validation_context*/, bool is_server,
+    const CertValidator::ExtraValidationContext& validation_context, bool is_server,
     absl::string_view host_name) {
   ASSERT(callback != nullptr);
   ASSERT(callback_ == nullptr);
   callback_ = std::move(callback);
   if (expected_host_name_.has_value()) {
     EXPECT_EQ(expected_host_name_.value(), host_name);
   }
+  if (expected_local_address_.has_value() && validation_context.local_address != nullptr) {
```

Comment:
Should it also expect validation_context.local_address is not nullptr in this case?

Reply:
Those two are a bit orthogonal in theory and local_address is added as optional field (i.e. nullptr).
When callbacks in `context_impl.h` is nullptr for whatever reasons I haven't run into, the disconnected situation happens. In other words, if such case happens in the future(which is possible that callbacks is nullptr because it is returned by `CRYPTO_get_ex_data`), the test will fail if we tie those two here. 

So on safe side, i check nullptr.

Reply:
What if local_address is not at all?

Reply:
Please see my reply above. 
I can change it to assert or check it here as localAddress is probably not nullptr in practice. But the goal of this PR is adding **optional** field that is only required by specific extension. It is also the reason I changed it to sharePtr pointer based on previous comments.

So here is really the nits I don't have preference. But I want to choose the safest approach as it is an optional field .

Reply:
Does this test prevent regression if someone revert your change in ContextImpl?

Reply:
well, if people want to revert the change, they can just remove the test as well.
but i don't mind to add a small blocker by adding an assert.